File: app/src/main/java/com/project/mvpframe/widget/NoDataView.java
package com.project.mvpframe.widget;

import android.annotation.TargetApi;
import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.FrameLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.project.mvpframe.R;


/**
 * @CreateDate 2019/12/18 10:03
 * @Author jaylm
 */
public class NoDataView extends FrameLayout {

    public NoDataView(@NonNull Context context) {
        this(context, null);

    }

    public NoDataView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public NoDataView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        this(context, attrs, defStyleAttr, 0);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public NoDataView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        LayoutInflater.from(getContext()).inflate(R.layout.layout_no_data, this);
    }

    public NoDataView setText(CharSequence charSequence) {
        TextView tvNotice = findViewById(R.id.tv_notice);
        tvNotice.setText(charSequence);
        return this;
    }
}

File: app/src/main/java/com/project/mvpframe/widget/NoDataView.kt
package com.project.mvpframe.widget

import android.annotation.TargetApi
import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.FrameLayout
import android.widget.TextView

import com.project.mvpframe.R


/**
 * @CreateDate 2019/12/18 10:03
 * @Author jaylm
 */
class NoDataView @TargetApi(Build.VERSION_CODES.LOLLIPOP)
constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
    FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
    @JvmOverloads
    constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : this(context, attrs, defStyleAttr, 0)

    init {
        LayoutInflater.from(getContext()).inflate(R.layout.layout_no_data, this)
    }

    fun setText(charSequence: CharSequence): NoDataView {
        val tvNotice = findViewById<TextView>(R.id.tv_notice)
        tvNotice.text = charSequence
        return this
    }
}
